### PR TITLE
fix: preserve supplementary plane characters (U+10000+) in bin_xml_escape

### PR DESCRIPTION
**Pull request by thakoreh**
### Problem
`bin_xml_escape` in `src/_pytest/junitxml.py` incorrectly escapes all supplementary plane characters (U+10000 and above). This includes:
- All emoji (😀, 🎉, etc.)
- CJK Extension B-F characters
- Mathematical symbols (𝐀, 𝐁, etc.)
- Musical symbols (𝄞)

That is ~1,048,576 valid XML characters being treated as illegal.

### Root Cause
The regex `\u10000-\u10ffff` uses 4-digit `\u` escapes, so Python parses `\u10000` as `\u1000` (Myanmar letter KA, U+1000) + literal `0`, and `\u10ffff` as `\u10ff` + literal `ff`.

### Fix
Changed `\u10000-\u10ffff` to `\U00010000-\U0010ffff` (8-digit `\U` escapes).

### Testing
- Added `test_bin_xml_escape_supplementary_plane_chars` covering emoji, CJK Extension B, musical/mathematical symbols, and boundary values
- All 136 existing junitxml tests continue to pass
- Verified the fix preserves supplementary plane chars while still escaping truly illegal characters

```python
# Before fix
bin_xml_escape("test_😀_passes")  # → "test_#x1F600_passes" ❌

# After fix
bin_xml_escape("test_😀_passes")  # → "test_😀_passes" ✅
```

This is a regression from commit 1653c49b1b which refactored from dynamic `chr()` to static string.

### CODE REVIEW

File: testing/test_junitxml.py [hunk 1]
```diff
@@ -1841,3 +1831,51 @@ def test_no_message_quiet(pytester: Pytester) -> None:
 
     result = pytester.runpytest("--junitxml=pytest.xml", "--quiet")
     result.stdout.no_fnmatch_line("* generated xml file: *")
+
+
+class TestBinXmlEscapeSupplementaryPlane:
```
Comment (bluetech) [inline testing/test_junitxml.py:1836]: These tests seem a bit too excessive, perhaps update the existing `test_invalid_xml_escape` test to have proper coverage of the high range?

**Review by bluetech — COMMENTED** [review]
Thanks. Please see my comment, and also add a short changelog entry in `changelog/14483.bugfix.rst` and should be good to merge.

**Comment by thakoreh** [comment]
Closing this while I pause open-source contributions. Apologies for the noise.
